code/modules/admin/sql_notes.dm

```diff
@@ -0,0 +1,186 @@
+/proc/add_note(target_ckey, notetext, timestamp, adminckey, logged = 1, server)
+	if(!check_rights(R_INVESTIGATE, 1, usr)) return
+
+	if(!dbcon.IsConnected())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+
+	if(!target_ckey)
+		var/new_ckey = ckey(input(usr,"Who would you like to add a note for?","Enter a ckey",null) as text)
+		if(!new_ckey)
+			return
+		new_ckey = sanitizeSQL(new_ckey)
+		var/DBQuery/query_find_ckey = dbcon.NewQuery("SELECT ckey FROM erro_player WHERE ckey = '[new_ckey]'")
+		if(!query_find_ckey.Execute())
+			var/err = query_find_ckey.ErrorMsg()
+			log_game("SQL ERROR obtaining ckey from player table. Error : \[[err]\]\n")
+			return
+		if(!query_find_ckey.NextRow())
+			if(alert(usr, "[new_ckey] has not been seen before, are you sure you want to add them to the watchlist?", "Unknown ckey", "Yes", "No", "Cancel") != "Yes")
+				return
+		target_ckey = new_ckey
+	var/target_sql_ckey = sanitizeSQL(target_ckey)
+	if(!notetext)
+		notetext = input(usr,"Write your Note","Add Note") as message
+		if(!notetext)
+			return
+	notetext = sanitizeSQL(notetext)
+	if(!timestamp)
+		timestamp = SQLtime()
+	if(!adminckey)
+		adminckey = usr.ckey
+		if(!adminckey)
+			return
+	var/admin_sql_ckey = sanitizeSQL(adminckey)
+	if(!server)
+		if (config && config.server_name)
+			server = config.server_name
+	server = sanitizeSQL(server)
+	var/DBQuery/query_noteadd = dbcon.NewQuery("INSERT INTO erro_messages (type, targetckey, adminckey, text, timestamp, server) VALUES ('note', '[target_sql_ckey]', '[admin_sql_ckey]', '[notetext]', '[timestamp]', '[server]')")
+	if(!query_noteadd.Execute())
+		var/err = query_noteadd.ErrorMsg()
+		log_game("SQL ERROR adding new note to table. Error : \[[err]\]\n")
+		return
+	if(logged)
+		log_admin("[key_name(usr)] has added a note to [target_ckey]: [notetext]")
+		message_admins("[key_name_admin(usr)] has added a note to [target_ckey]:<br>[notetext]")
+		send2adminirc("[get_key(usr)] has added a note to [target_ckey]: \"[notetext]\"")
+		show_note(target_ckey)
+
+/proc/remove_note(note_id)
+	var/ckey
+	var/notetext
+	var/adminckey
+
+	if(!check_rights(R_INVESTIGATE, 1, usr)) return
+
+	if(!dbcon.IsConnected())
+		to_chat(usr, "<span class='danger'>Failed to establish database connection.</span>")
+		return
+	if(!note_id)
+		return
+	note_id = text2num(note_id)
+	var/DBQuery/query_find_note_del = dbcon.NewQuery("SELECT targetckey, text, adminckey FROM erro_messages WHERE id = [note_id] and type = 'note'")
```

Comment:
dbcon.NewQuery("SELECT \`targetckey\`, \`text\`, \`adminckey\` FROM \`erro_messages\` WHERE \`id\` = [note_id] and \`type\` = 'note'")